website/status_screen/templates/status_screen/venue_music_screen.html

```diff
@@ -0,0 +1,234 @@
+                <a class="navbar-brand drop-out-header" href="/"><img
+                        src="{% static 'tosti/svg/TOSTI-logo.svg' %}"
+                        height="120"/>
+                </a>
+            </div>
+        </div>
+    </nav>
+{% endblock %}
+
+{% block page %}
+    <div class="d-flex flex-column justify-content-center align-items-center mt-5" id="music-screen-container">
+        <h1 class="mt-5">Currently playing:</h1>
+        <div class="player" id="player-container" style="width: 65%">
+        <template v-if="player_data.track.name !== null">
+        <div class="w-50 mx-auto mb-2" v-if="player_data.track.image !== null">
+            <a href="{% url "thaliedje:now_playing" player=player %}">
+                <img class="w-100" :src="player_data.track.image" :alt="player_data.track.name"/>
+            </a>
+        </div>
+        <div class="w-50 mx-auto mb-3" v-if="this.track_progress_ms !== null && this.track_progress_percentage !== null && this.player_data !== null && this.player_data.duration_ms !== null">
+            <div class="progress" style="height: 10px">
+                <div
+                    class="progress-bar"
+                    role="progressbar"
+                    :style="'width: ' + this.track_progress_percentage+'%;'"
+                    :aria-valuenow="this.track_progress_ms"
+                    aria-valuemin="0"
+                    :aria-valuemax="this.player_data.duration_ms">
+                </div>
+                </div>
+            <div>
+                <h4 class="float-start me-2 align-left text-nowrap" style="width: 30px">
+                    ${ Math.floor((track_progress_ms / 1000) / 60) }$:${ String(Math.floor((track_progress_ms / 1000) % 60)).padStart(2, '0') }$
+                </h4>
+                <h4 class="float-end ms-2 align-right">
+                    ${ Math.floor((player_data.duration_ms / 1000) / 60) }$:${ String(Math.floor((player_data.duration_ms / 1000) % 60)).padStart(2, '0') }$
+                    <!--(${ -1 * Math.floor((track_ms_left / 1000) / 60) }$:${ String(Math.floor((track_ms_left / 1000) % 60)).padStart(2, '0') }$) -->
+                </h4>
+            </div>
+        </div>
+        <h2 class="text-center m-auto mt-5">${ player_data.track.name }$</h2>
+        <div class="text-center w-100">
+            <h3>
+                <template v-for="(artist, index) in player_data.track.artists">
+                    <template v-if="index === player_data.track.artists.length - 1">
+                        ${ artist }$
+                    </template>
+                    <template v-else>
+                        ${ artist }$ -
+                    </template>
+                </template>
+            </h3>
+        </div>
+        </template>
+        <template v-else>
+            <h1 class="text-center m-auto">No currently playing track</h1>
+        </template>
+    </div>
```

Comment:
Can't we simply include the player card here? We have template_tags to render a player...

Reply:
Yeah I saw that and I tried, but since my page contains vue code and the template does as well this does not go well. The vue on my page skips over the script that contains the vue code for the template tag. And I get the following error:

![image](https://github.com/user-attachments/assets/194a4b48-7b98-45e9-b052-101aab4e4c3f)

So I tried to copy as little as I need but I will try to look for a better alternative since this is a lot of double code.

Reply:
Nvm I might just be a fucking idiot and I should put the template outside of the component I mount the other vue on...